src/components/views/dialogs/security/AccessSecretStorageDialog.tsx

```diff
@@ -222,34 +222,44 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
         // rest of this function.
         Modal.toggleCurrentDialogVisibility();
 
-        // Force reset secret storage (which resets the key backup)
-        await accessSecretStorage(async () => {
-            // Now reset cross-signing so everything Just Works™ again.
-            const cli = MatrixClientPeg.get();
-            await cli.bootstrapCrossSigning({
-                authUploadDeviceSigningKeys: async (makeRequest) => {
-                    // XXX: Making this an import breaks the app.
-                    const InteractiveAuthDialog = sdk.getComponent("views.dialogs.InteractiveAuthDialog");
-                    const { finished } = Modal.createTrackedDialog(
-                        'Cross-signing keys dialog', '', InteractiveAuthDialog,
-                        {
-                            title: _t("Setting up keys"),
-                            matrixClient: cli,
-                            makeRequest,
+        try {
+            // Force reset secret storage (which resets the key backup)
+            await accessSecretStorage(async () => {
+                try {
```

Comment:
Why is this inner one needed...? `accessSecretStorage` re-throws an errors that happen inside it, so they should all bubble up to your outer catch...

Reply:
Oh, does it? Neat :D